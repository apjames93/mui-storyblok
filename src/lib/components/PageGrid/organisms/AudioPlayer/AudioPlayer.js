import React from 'react';
import PropTypes from 'prop-types';
import AudioPlayer from 'material-ui-audio-player';
import {
  dimensionProp,
  muiStringProp,
  muiBlokNumberProp,
} from 'lib/utils/customProps';

/**
 * https://www.npmjs.com/package/material-ui-audio-player
 */

const MuiAudioPlayer = ({
  src,
  width,
  variation,
  download,
  autoplay,
  elevation,
  rounded,
  spacing,
  order,
  loop,
  preload,
  dataBlokC,
  dataBlokUid,
  storyblokClass,
}) => (
  <AudioPlayer
    src={src}
    width={width}
    variation={variation}
    download={download}
    autoplay={autoplay}
    elevation={elevation}
    rounded={rounded}
    spacing={spacing}
    order={order}
    loop={loop}
    preload={preload}
    className={storyblokClass}
    data-blok-c={dataBlokC}
    data-blok-uid={dataBlokUid}
  />
);

export default MuiAudioPlayer;

MuiAudioPlayer.propTypes = {
  /**
     * Could accept audio link or array of audio links.
     */
  src: PropTypes.string.isRequired,
  /**
   * Corresponds to style property width.
   */
  width(props, propName, componentName) {
    return dimensionProp(props, propName, componentName);
  },
  /**
   * Component view variation.
   * options: default, primary, secondary
   */
  variation(props, propName, componentName) {
    const validProp = ['default', 'primary', 'secondary'];
    return muiStringProp(props, propName, componentName, validProp);
  },
  /**
   * Display download button (icon) with dropdown of available audio tracks for download.
   */
  download: PropTypes.bool,
  /**
   * Display download button (icon) with dropdown of available audio tracks for download.

   */
  autoplay: PropTypes.bool,
  /**
     * Shadow depth. Corresponds to elevation prop of Material Ui Paper component.
     */
  elevation: PropTypes.number,
  /**
   * Rounded corners of the container. Corresponds to square prop of Material Ui Paper component.
   */
  rounded: PropTypes.bool,
  /**
   * Spacing for root Grid container. Corresponds to spacing prop of Material Ui Grid component.
   */
  spacing(props, propName, componentName) {
    const validProps = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
    return muiBlokNumberProp(props, propName, componentName, validProps);
  },
  /**
   * Order of Slider and controls buttons.
    * options: standart, reverse
   */
  order(props, propName, componentName) {
    const validProp = ['standart', 'reverse'];
    return muiStringProp(props, propName, componentName, validProp);
  },
  /**
   * Display loop button.
   */
  loop: PropTypes.bool,
  /**
 * Corresponds to HTML audio attribute preload.
 * options auto|metadata|none
 */
  preload(props, propName, componentName) {
    const validProp = ['auto', 'metadata', 'none'];
    return muiStringProp(props, propName, componentName, validProp);
  },

  /** storyblok prop for when in editor to allow click bridge */
  dataBlokC: PropTypes.string,
  /** storyblok prop for when in editor to allow click bridge */
  dataBlokUid: PropTypes.string,
  /** storyblok prop for when in editor to allow click bridge */
  storyblokClass: PropTypes.string,
};

MuiAudioPlayer.defaultProps = {
  width: '100%',
  variation: 'default',
  download: false,
  autoplay: false,
  elevation: 1,
  rounded: false,
  spacing: 3,
  loop: false,
  order: 'standart',
  preload: 'none',
  dataBlokC: '',
  dataBlokUid: '',
  storyblokClass: '',
};
