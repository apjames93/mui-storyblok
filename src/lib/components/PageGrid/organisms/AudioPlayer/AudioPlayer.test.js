import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import AudioPlayer from './AudioPlayer';

function setup() {
  const props = {
    src: 'http://thebornreadies.rocks/songs/Automatic.mp3',
  };
  const comp = shallow(<AudioPlayer {...props} />);
  return { comp, props };
}

describe('<AudioPlayer />', () => {
  it('renders AudioPlayer', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(<AudioPlayer {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
