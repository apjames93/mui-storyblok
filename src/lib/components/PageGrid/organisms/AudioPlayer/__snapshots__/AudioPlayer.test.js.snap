// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AudioPlayer /> snapshot 1`] = `
<div
  className="MuiPaper-root MuiGrid-root makeStyles-root-1 makeStyles-root-34 makeStyles-elevation1-10 makeStyles-rounded-8 MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-align-items-xs-center MuiPaper-elevation1 MuiPaper-rounded"
>
  <audio
    hidden={true}
    preload="none"
  >
    <source
      src="http://thebornreadies.rocks/songs/Automatic.mp3"
    />
  </audio>
  <div
    className="MuiGrid-root makeStyles-commonContainer-5 MuiGrid-item"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root makeStyles-icon-39 makeStyles-icon-40 MuiSvgIcon-fontSizeLarge"
      focusable="false"
      onClick={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M0 0h48v48H0z"
        fill="none"
        transform="scale(0.5, 0.5)"
      />
      <path
        d="M24 4C12.95 4 4 12.95 4 24s8.95 20 20 20 20-8.95 20-20S35.05 4 24 4zm-4 29V15l12 9-12 9z"
        transform="scale(0.5, 0.5)"
      />
    </svg>
  </div>
  <div
    className="MuiGrid-root makeStyles-commonContainer-43 makeStyles-volumeIconContainer-45 MuiGrid-item"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root makeStyles-icon-44 makeStyles-icon-48 MuiSvgIcon-fontSizeLarge"
      focusable="false"
      onClick={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M3 9v6h4l5 5V4L7 9H3zm13.5 3c0-1.77-1.02-3.29-2.5-4.03v8.05c1.48-.73 2.5-2.25 2.5-4.02zM14 3.23v2.06c2.89.86 5 3.54 5 6.71s-2.11 5.85-5 6.71v2.06c4.01-.91 7-4.49 7-8.77s-2.99-7.86-7-8.77z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </div>
  <div
    className="MuiGrid-root makeStyles-sliderContainerWrapper-2 makeStyles-sliderContainerWrapper-35 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item"
  >
    <div
      className="MuiGrid-root makeStyles-commonContainer-5 MuiGrid-item"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
        onClick={[Function]}
      >
        00:00
      </p>
    </div>
    <div
      className="MuiGrid-root makeStyles-sliderContainer-3 MuiGrid-item"
    >
      <span
        className="MuiSlider-root MuiSlider-colorPrimary makeStyles-slider-4 makeStyles-slider-36"
        onMouseDown={[Function]}
      >
        <span
          className="MuiSlider-rail"
        />
        <span
          className="MuiSlider-track"
          style={
            Object {
              "left": "0%",
              "width": "0%",
            }
          }
        />
        <input
          type="hidden"
          value="0"
        />
        <span
          aria-orientation="horizontal"
          aria-valuemax={100}
          aria-valuemin={0}
          aria-valuenow={0}
          className="MuiSlider-thumb MuiSlider-thumbColorPrimary"
          data-index={0}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          role="slider"
          style={
            Object {
              "left": "0%",
            }
          }
          tabIndex={0}
        />
      </span>
    </div>
    <div
      className="MuiGrid-root makeStyles-commonContainer-5 MuiGrid-item"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        00:00
      </p>
    </div>
  </div>
</div>
`;
